docs/state/substates.md

```diff
@@ -5,63 +5,109 @@ from pcweb.templates.docpage import docdemo_from
+The code below shows two states, `RectangleState` and `SquareState`, each representing metadata about rectangles and squares, along with the calculation of their respective areas.
 
 ```python exec
-class ParentState(State):
-    checked: bool = True
-    count: int = 0
+class RectangleState(rx.State):
+    name: str = "rectangle"
+    shape_type: str = "quadrilateral"
+    length: int = 10
+    width: int = 5
 
+    @rx.var
+    def area(self) -> int:
+        return self.length * self.width 
 
-class ChildState1(ParentState):
-    value: int = 42
+class SquareState(rx.State):
+    name: str = "square"
+    shape_type: str = "quadrilateral"
+    length: int = 4
 
+    @rx.var
+    def area(self) -> int:
+        return self.length ** 2
 
-class ChildState2(ParentState):
-    color: str = "red"
-
-
-class ChildState3(ChildState1):
-    text: str = "Hello World"
 
 
 def my_badge():
-    return rx.badge(ChildState3.text, color_scheme=ChildState2.color)
+    return rx.vstack(
+        rx.hstack(
+            rx.text(RectangleState.name),
+            rx.text(RectangleState.area),
+        ),
+        rx.hstack(
+            rx.text(SquareState.name),
+            rx.text(SquareState.area),
+        ),
+    )
 ```
 
 ```python eval
 docdemo_from(
-    ParentState,
-    ChildState1,
-    ChildState2,
-    ChildState3,
+    RectangleState,
+    SquareState,
     component=my_badge,
 )
 ```
 
-In the example above, we have a base state, `ParentState`, with two substates
-`ChildState1` and `ChildState2`. Additionally, `ChildState3` inherits from
-`ChildState1`. Components can access any var or event handler from any substate.
 
-A common use case may be to create a substate for each page of your app, while
-keeping general vars such as the logged in user in the base state for easy
-access.
+
+In this code, you can see that both states share a common resource (`shape_type`). We can improve code organization and promote reusability by introducing the concept of substates. 
+This involves creating a base state (`QuadrilateralState`), from which other states can inherit or subclass. After refactoring, the code takes on the following structure:
+
+```python
+class QuadrilateralState(rx.State):
+    shape_type: str = "quadrilateral"
+
+class RectangleState(QuadrilateralState):
+    name: str = "rectangle"
+    length: int = 10
+    width: int = 5
+
+    @rx.var
+    def area(self) -> int:
+        return self.length * self.width 
+
+class SquareState(QuadrilateralState):
+    name: str = "square"
+    length: int = 4
+
+    @rx.var
+    def area(self) -> int:
+        return self.length ** 2
+
+```
+
+It's important to note that certain variables, such as `length` and `name`, were not moved into the `QuadrilateralState` as they are not shared among substates. 
+This approach streamlines the code, ensuring that only common resources are centralized in the base state while preserving the distinct characteristics of individual substates.
```

Comment:
I think maybe we can have two separate sections. The first section shows just subclassing from `rx.State` for different pages. Then a second example can show subclassing another substate, which will then have this feature of they can share certain variables.